sample/sample-with-di/helloworldfeature/src/main/java/com/dropbox/kaiken/sample/advancedsample/helloworldfeature/LoginScreen.kt

```diff
@@ -0,0 +1,108 @@
+package com.dropbox.kaiken.sample.advancedsample.helloworldfeature
+
+import android.content.Context
+import android.content.Intent
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.Column
+import androidx.compose.material.Button
+import androidx.compose.material.MaterialTheme
+import androidx.compose.material.Text
+import androidx.compose.material.TextField
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.LocalContext
+import com.dropbox.kaiken.skeleton.core.SkeletonOauth2
+import com.dropbox.kaiken.skeleton.scoping.AuthOptionalScreenScope
+import com.dropbox.kaiken.skeleton.scoping.SingleIn
+import com.squareup.anvil.annotations.ContributesMultibinding
+import javax.inject.Inject
+
+class UserInput(val userId: String)
+abstract class LoginPresenter :
+    Presenter<LoginPresenter.LoginEvent, LoginPresenter.LoginModel>(LoginNeeded) {
+    sealed interface LoginEvent
+    data class Submit(val userInput: UserInput) : LoginEvent
+
+    sealed interface LoginModel
```

Comment:
Not a functional comment about the injection pattern, but being this being a sample implementation of a presenter pattern the approach taken for Models (or UiState) seems a bit weird, to me.  These models feel a bit more like a stream of events telling the view which state to move to, as opposed to a model object representing the state of the view. 

So instead of a bunch fo individual models for each state of the view, I'd expect something like this:

```kotlin
LoginModel(
  val isLoading: Boolean = false,
  val rememberChecked: Boolean = false,
  val username: String = "", // These two could be optional, assuming there's no prepopulation of usernames
  val password: String = "",
)
```

This allows the Model to represent the UI state, and allows the and allows certain parts of the view to be updated without losing the state of the rest of the view (i.e. login fails, so you go back to login needed, but you no longer know if remember me should be checked unless you keep the state history).

Reply:
yeah I think as the sample grows this is what it will become, more of a wide state.